container-host-files/etc/hcf/config/scripts/fetch_uaa_verification_key.sh

```diff
@@ -7,19 +7,50 @@
 
 # Note that this is *sourced* into run.sh, so we can't exit the shell.
 
-# Wait for UAA
-while true ; do
-    if curl --fail \
-        $(if test "${SKIP_CERT_VERIFY_EXTERNAL}" = "true" ; then echo "--insecure" ; fi) \
-        "${HCF_UAA_INTERNAL_URL}/token_key"
-    then
-        break
-    fi
-    sleep 10
-done
+
+# Report progress to the user; use as printf
+status() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;32m" "$@" "\033[0m"
+}
+
+# Report problem to the user; use as printf
+trouble() {
+    local fmt="${1}"
+    shift
+    printf "\n%b${fmt}%b\n" "\033[0;31m" "$@" "\033[0m"
+}
+
+# helper function to retry a command several times, with a delay between trials
+# usage: retry <max-tries> <delay> <command>...
+function retry_forever () {
+    delay=${1}
+    shift 1
+
+    while true ; do
+	printf "Trying: %s\n" "$*"
+        if "$@" ; then
+	    status ' SUCCESS'
+            break
+        fi
+	trouble '  FAILED'
+	status "Waiting ${delay} ..."
+        sleep "${delay}"
+    done
+}
+
+SKIP=$(if test "${SKIP_CERT_VERIFY_EXTERNAL}" = "true" ; then echo "--insecure" ; fi)
+
+status "Waiting for UAA to be available at ${HCF_UAA_INTERNAL_URL}/token_key ..."
+retry_forever 10s curl --connect-timeout 5 --fail $SKIP "${HCF_UAA_INTERNAL_URL}/token_key"
```

Comment:
`retry_forever` now repeatedly prints out the command (which is the URL), so we probably don't need it in the status.